controllers/topics/checkName.js

```diff
@@ -0,0 +1,37 @@
+const {Topics} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.query;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    let response = {
+      success: false,
+      message: 'A community with this name already exists',
+      available: false
+    };
+
+    if (!topics) {
+      response = {
+        success: true,
+        message: 'Community name available',
+        available: true
+      };
+    }
+
+    return res.status(200).json(response);
+  } catch (error) {
+    return res.json({
+      code: error.statusCode,
+      status: 'fail',
+      message: error.message,
+      data: 'null'
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic of the code seems fine, but there's a potential security issue. You are directly using the user input from `req.query` without any sanitization or validation. This could lead to SQL injection attacks if an attacker passes in malicious SQL statements as part of the `name` query parameter.

To mitigate this, you should sanitize and validate all user inputs before using them. Here's how you can do it:

```diff
  try {
    let {name} = req.query;
+    if (!name || typeof name !== 'string') {
+      return res.status(400).json({
+        success: false,
+        message: 'Invalid community name',
+      });
+    }
    name = name.toLowerCase();
```

Also, consider using HTTP status codes to indicate the type of error. For example, use 400 for bad requests (invalid input) and 500 for internal server errors. In your catch block, you're always returning a 200 status code, even when there's an error. This might confuse clients because they'll see a 200 status code (which usually means success) but the response body will contain an error message.

Here's how you can improve the error handling:

```diff
    return res.json({
+    return res.status(error.statusCode || 500).json({
      code: error.statusCode,
      status: 'fail',
      message: error.message,
      data: 'null'
    });
```
This way, the client will receive an appropriate HTTP status code depending on the type of error.


<!-- This is an auto-generated comment by OSS CodeRabbit -->